speaker/layer2_controller.go

```diff
@@ -133,9 +142,20 @@ func (c *layer2Controller) ShouldAnnounce(l log.Logger, name string, toAnnounce
 	return "notOwner"
 }
 
+func getInterfaces(localNode string, l2Advertisements []*config.L2Advertisement) []string {
+	ifs := sets.NewString()
+	for _, l2 := range l2Advertisements {
+		if matchNode, ok := l2.Nodes[localNode]; !ok || !matchNode {
+			continue
+		}
+		ifs = ifs.Insert(l2.Interfaces...)
+	}
+	return ifs.List()
+}
+
 func (c *layer2Controller) SetBalancer(l log.Logger, name string, lbIPs []net.IP, pool *config.Pool) error {
 	for _, lbIP := range lbIPs {
-		c.announcer.SetBalancer(name, lbIP)
+		c.announcer.SetBalancer(name, lbIP, pool.Name)
```

Comment:
Assuming you maintain the mapping between pool name and interface outside, here you can just pass the list of interfaces.

Reply:
ok